src/darwin/Framework/CHIP/MTRBaseDevice.h

```diff
@@ -584,12 +584,55 @@ API_AVAILABLE(ios(16.4), macos(13.3), watchos(9.4), tvos(16.4))
 
 @interface MTRAttributeReport : NSObject
 @property (nonatomic, readonly, copy) MTRAttributePath * path;
-// value is nullable because nullable attributes can have nil as value.
+
+/**
+ * value will be nil in the following cases:
+ *
+ * * There was an error.  In this case, "error" will not be nil.
+ * * The attribute is nullable and the value of the attribute is null.
+ *
+ * If value is not nil, the actual type of value will depend on the
+ * schema-defined (typically defiend in the Matter specification) type of the
+ * attribute as follows:
+ *
+ * * list: NSArray of whatever type the list entries are.
+ * * struct: The corresponding structure interface defined by Matter.framework
+ * * octet string: NSData
+ * * string: NSString
+ * * discrete/analog types: NSNumber
+ *
+ * Derived types are represented as the base type, except for "string".
```

Comment:
Not sure I understand this last sentence

Reply:
In the spec, "string" is a derived type of "octet string", but they are represented as different types here.

I'll expand this out.

Reply:
@jake-apple https://github.com/project-chip/connectedhomeip/pull/26321/commits/84444cbbafa1b62e837a25b71a5b75cf689f8a8a